src/components/panel/panel.js

```diff
@@ -527,5 +531,16 @@ MdPanelRef.prototype._createPanel = function() {
         if (self._config['panelClass']) {
           self._panelEl.addClass(self._config['panelClass']);
         }
+
+        self._addStyles();
       });
 };
+
+
+/**
+ * Adds the styles for the panel, such as positioning and z-index.
+ * @private
+ */
+MdPanelRef.prototype._addStyles = function() {
+  this._panelContainer.css('z-index', this._config['zIndex'] || defaultZIndex);
```

Comment:
If `config['zIndex'] == 0` it will go to the default index, when it's technially possible that someone would want to set a `0` index.

Instead you could do something like
```js
var configZindex = this._config['zIndex'];
this._panelContainer.css('z-index', configZIndex != null ? configZIndex : defaultZIndex);
```